website/docs/script/spawn_direct_cross-script_calling.mdx

```diff
@@ -0,0 +1,322 @@
+    OtherEndClosed,
+    /// Max vms has been spawned. Its value is 8.
+    MaxVmsSpawned,
+    /// Max fds has been created. Its value is 9.
+    MaxFdsCreated,
+}
+```
+
+Among these errors, a notable one is OtherEndClosed. It allows us to implement a function like `read_all(fd) -> Vec<u8>` to read all the data from the pipe at once. Many programming languages provide similar functions in their standard libraries, and you can easily implement this functionality yourself in the CKB-VM.
+
+```rust
+pub fn read_all(fd: u64) -> Result<Vec<u8>, SysError> {
+    let mut ret = Vec::new();
+    let mut buf = [0; 256];
+    loop {
+        match syscalls::read(fd, &mut buf) {
+            Ok(data) => ret.extend_from_slice(&buf[..data]),
+            Err(SysError::OtherEndClosed) => break,
+            Err(err) => return Err(err),
+        }
+    }
+    Ok(ret)
+}
+```
+
+## Example
+
+Let's walk through a basic example of using spawn. Imagine we want to deploy a public library on the chain, which is quite simple: it receives input parameters, concatenates them, and returns the result to the caller. Let’s see how to implement it.
+
+The main code only consists of the following four lines, where:
+
+- The first line obtains the file descriptor.
+- The second line gathers and concatenates all the arguments.
+- The third line writes the result to the file descriptor.
+- The last line closes the file descriptor.
+
+```rust
+#[no_mangle]
+pub unsafe extern "C" fn main() -> u64 {
+    let fds = inherited_fds();
+    let out = ARGS.join("");
+    write(fds[0], out.as_bytes());
+    close(fds[0]);
+    return 0;
+}
+```
+
+So, how to use this library? Follow the following code:
+
+- The first line defines the arguments to be passed to the child Script.
+- The second line creates a pipe.
+- The third line calls the `spawn()` function, passing in the arguments and the writable file descriptor.
+- The fourth line reads all the data from the readable file descriptor.
+- The last line uses the `wait()` function to wait until the child Script exits.
+
+```rust
+#[no_mangle]
+pub unsafe extern "C" fn main() -> u64 {
+    let argv = ["Hello", "World!"];
+    let fds = pipe();
+    let pid = spawn_cell(code_hash, hash_type, &argv, &[fds[1]]);
+    let data = read_all(fds[0]);
+    assert_eq!(&data, b"Hello World!");
+    wait(pid);
+    return 0;
+}
+```
+
+## Enhanced Efficiency With Spawn’s Pipes
+
+Since `spawn()` operates within its own memory space, independent of the parent process, it can mitigate certain security risks, particularly in scenarios involving sensitive data.
+
+While some other blockchain virtual machines, such as EVM, also offer cross-contract call functions, these calls tend to be quite expensive, especially when invoking child contracts repeatedly. This is primarily because it requires continuously creating and destroying a virtual machine to complete each call, pictured as below:
```

Comment:
```suggestion
While some other blockchain virtual machines, such as the EVM, also offer cross-contract call functions, these calls tend to be quite expensive, especially when invoking child contracts repeatedly. This is primarily because each call requires continuously creating and destroying a virtual machine, as illustrated below:
```